src/libslic3r/PrintConfig.cpp

```diff
@@ -2321,32 +2321,121 @@ void PrintConfigDef::init_fff_params()
     def->set_default_value(new ConfigOptionStrings{L("(Undefined)")});
     def->cli = ConfigOptionDef::nocli;
 
-    def = this->add("infill_direction", coFloat);
+    def = this->add("infill_direction", coFloatOrPercent);
     def->label = L("Sparse infill direction");
     def->category = L("Strength");
-    def->tooltip = L("Angle for sparse infill pattern, which controls the start or main direction of line");
+    def->tooltip = L("Angle for sparse infill pattern, which controls the start or main direction of line.");
     def->sidetext = "°";
     def->min = 0;
     def->max = 360;
+    def->max_literal = 10;
     def->mode = comAdvanced;
-    def->set_default_value(new ConfigOptionFloat(45));
+    def->set_default_value(new ConfigOptionFloatOrPercent(45., false));
 
-    def = this->add("solid_infill_direction", coFloat);
+    def = this->add("solid_infill_direction", coFloatOrPercent);
     def->label = L("Solid infill direction");
     def->category = L("Strength");
-    def->tooltip = L("Angle for solid infill pattern, which controls the start or main direction of line");
+    def->tooltip = L("Angle for solid infill pattern, which controls the start or main direction of line.");
     def->sidetext = "°";
     def->min = 0;
     def->max = 360;
+    def->max_literal = 10;
     def->mode = comAdvanced;
-    def->set_default_value(new ConfigOptionFloat(45));
+    def->set_default_value(new ConfigOptionFloatOrPercent(45., false));
 
-    def = this->add("rotate_solid_infill_direction", coBool);
-    def->label = L("Rotate solid infill direction");
+    def = this->add("top_infill_direction", coFloatOrPercent);
+    def->label = L("Top infill direction");
     def->category = L("Strength");
-    def->tooltip = L("Rotate the solid infill direction by 90° for each layer.");
+    def->tooltip = L("Angle for top infill pattern, which controls the start or main direction of line" ".\n"
+                    "If expressed as a %, it will be computed a sector of the full turn 360°" ".\n");
+    def->sidetext = "° or %";
+    def->min = -360;
+    def->max = 360;
+    def->max_literal = 10;
     def->mode = comAdvanced;
-    def->set_default_value(new ConfigOptionBool(true));
+    def->set_default_value(new ConfigOptionFloatOrPercent(45., false));
+
+    def = this->add("bottom_infill_direction", coFloatOrPercent);
+    def->label = L("Bottom infill direction");
+    def->category = L("Strength");
+    def->tooltip = L("Angle for bottom infill pattern, which controls the start or main direction of line" ".\n"
+                    "If expressed as a %, it will be computed a sector of the full turn 360°" ".\n");
+    def->sidetext = "° or %";
+    def->min = -360;
+    def->max = 360;
+    def->max_literal = 10;
+    def->mode = comAdvanced;
+    def->set_default_value(new ConfigOptionFloatOrPercent(45., false));
+
+    def = this->add("rotate_solid_infill_direction", coFloatOrPercent);
+    def->label = L("  Rotate solid infill");
```

Comment:
Why the spaces at the beginning of the string?

Reply:
Is this the correct way to indent? I don't see any other string indented like this... and I believe this alignment can break if the translated string is larger than the original.